savant/converter/yolo.py

```diff
@@ -56,92 +104,106 @@ def __call__(
-            class_ids = class_ids[class_mask]
-            confidences = confidences[class_mask]
-
-        # filter by confidence
-        if self.confidence_threshold:
-            conf_mask = confidences > self.confidence_threshold
-            bboxes = bboxes[conf_mask]
-            class_ids = class_ids[conf_mask]
-            confidences = confidences[conf_mask]
-
-        # TODO: ability to filter by size (width, height) and aspect ratio
-        # apply class agnostic NMS (all classes are treated as one)
-        if self.nms_iou_threshold > 0 and len(confidences) > 1:
-            nms_mask = nms_cpu(bboxes, confidences, self.nms_iou_threshold, self.top_k)
-            bboxes = bboxes[nms_mask]
-            class_ids = class_ids[nms_mask]
-            confidences = confidences[nms_mask]
-
-        # select top k
-        elif len(confidences) > self.top_k:
-            top_k_mask = np.argpartition(confidences, -self.top_k)[-self.top_k :]
-            bboxes = bboxes[top_k_mask]
-            class_ids = class_ids[top_k_mask]
-            confidences = confidences[top_k_mask]
-
-        roi_left, roi_top, roi_width, roi_height = roi
-
-        # scale
-        if model.input.maintain_aspect_ratio:
-            bboxes /= min(
-                model.input.width / roi_width,
-                model.input.height / roi_height,
+            mask = np.isin(class_ids, self.class_ids)
+            bboxes, confidences, class_ids = (
+                bboxes[mask],
+                confidences[mask],
+                class_ids[mask],
+            )
+
+        # --- confidence filter
+        if self.confidence_threshold > 0:
+            mask = confidences > self.confidence_threshold
+            bboxes, confidences, class_ids = (
+                bboxes[mask],
+                confidences[mask],
+                class_ids[mask],
+            )
+
+        # --- NMS (class-agnostic)
+        if self.nms_iou_threshold > 0 and bboxes.shape[0] > 1:
+            keep = nms_cpu(bboxes, confidences, self.nms_iou_threshold, self.top_k)
+            bboxes, confidences, class_ids = (
+                bboxes[keep],
+                confidences[keep],
+                class_ids[keep],
+            )
+        elif bboxes.shape[0] > self.top_k:
+            idx = np.argpartition(confidences, -self.top_k)[-self.top_k :]
+            bboxes, confidences, class_ids = (
+                bboxes[idx],
+                confidences[idx],
+                class_ids[idx],
             )
+
+        # --- map from model input space -> ROI space
+        roi_left, roi_top, roi_w, roi_h = roi
+
+        if model.input.maintain_aspect_ratio:
+            # reverse letterbox: remove padding, then de-scale
+            scale = min(model.input.width / roi_w, model.input.height / roi_h)
+            if scale <= 0:
+                return None
+            # de-scale w,h,xc,yc
+            bboxes /= scale
+
+            if getattr(model.input, 'symmetric_padding', False):
```

Comment:
Base model input has default value for symmetric_padding (as well as for maintain_aspect_ratio), there is no reason to use `getattr` here.